daemon/src/wire.rs

```diff
@@ -174,6 +175,10 @@ pub enum MakerToTaker {
         oracle_event_id: BitMexPriceEventId,
         tx_fee_rate: TxFeeRate,
         funding_rate: FundingRate,
+        /// Fee to be paid for the rollover.
```

Comment:
📣 We are not gonna be able to do rollovers with takers on a previous version. I think we need to introduce a new variant.

Reply:
Why not? An old taker who does not know about this field will just ignore it when deserializing the message.

Reply:
That is true.

But I think it is difficult to follow  some of the implicit logic introduced with this change. For example:

- Latest maker confirms rollover, sending over `complete_fee`.
- Old taker ignores `complete_fee`, goes into `setup_contract::roll_over`.
- Latest maker also goes into `setup_contract::roll_over` and uses the `complete_fee` passed in.
- Old taker calculates the `complete_fee` differently depending on the version.

I think by removing the rollover version handling code we run the risk of breaking old takers. I would have continued with adding a new version of the rollover protocol and a new confirmation message. That way we are explicitly aware of what is going on depending on what kind of taker we're dealing with. The code might be complex (several branches) but at least it's explicit.